packages/theme/src/types.ts

```diff
@@ -0,0 +1,121 @@
+import {TextStyle} from 'react-native'
+
+export type Theme = {
+  color: Palette
+  typography: Typography
+  spacing: Spacing
+}
+
+export type HexColor = `#${string}`
+
+export type Gradient = [HexColor, HexColor]
+
+export type Palette = {
+  'black-static': HexColor
+  'white-static': HexColor
+  'gray': {
+    max: HexColor
+    900: HexColor
+    800: HexColor
+    700: HexColor
+    600: HexColor
+    500: HexColor
+    400: HexColor
+    300: HexColor
+    200: HexColor
+    100: HexColor
+    50: HexColor
+    min: HexColor
+  }
+  'primary': {
+    900: HexColor
+    800: HexColor
+    700: HexColor
+    600: HexColor
+    500: HexColor
+    400: HexColor
+    300: HexColor
+    200: HexColor
+    100: HexColor
+  }
+  'secondary': {
+    900: HexColor
+    800: HexColor
+    700: HexColor
+    600: HexColor
+    500: HexColor
+    400: HexColor
+    300: HexColor
+    200: HexColor
+    100: HexColor
+  }
+  'magenta': {
+    700: HexColor
+    600: HexColor
+    500: HexColor
+    300: HexColor
+    100: HexColor
+  }
+  'cyan': {
+    500: HexColor
+    100: HexColor
+  }
+  'yellow': {
+    500: HexColor
+    100: HexColor
+  }
+  'gradients': {
+    'blue-green': Gradient
+    'green': Gradient
+    'blue': Gradient
+  }
+  'overlay-extension': {hex: HexColor; opacity: number}
```

Comment:
does it require to be in `hex` ?

Reply:
hmnm, let me double-check this later as I think we can remove some of it.. was not sure that we will use it